Source/NSGridView.m

```diff
@@ -0,0 +1,1280 @@
+      [coder encodeValueOfObjCType: @encode(CGFloat)
+                                at: &_leadingPadding];
+      [coder encodeObject: _gridView];
+      [coder encodeValueOfObjCType: @encode(CGFloat)
+                                at: &_trailingPadding];
+      [coder encodeValueOfObjCType: @encode(CGFloat)
+                                at: &_width];
+      [coder encodeValueOfObjCType: @encode(NSInteger)
+                                at: &_xPlacement];
+    }
+}
+
+- (instancetype) initWithCoder: (NSCoder *)coder
+{
+  self = [super init];
+  if (self != nil)
+    {
+      NSDebugLog(@"%@ %@",NSStringFromClass([self class]), NSStringFromSelector(_cmd));
+      // _cells = [[NSMutableArray alloc] initWithCapacity: 10];
+      if ([coder allowsKeyedCoding])
+        {
+          if ([coder containsValueForKey: @"NSGrid_hidden"])
+            {
+              _isHidden = [coder decodeBoolForKey: @"NSGrid_hidden"];
+            }
+          if ([coder containsValueForKey: @"NSGrid_leadingPadding"])
+            {
+              _leadingPadding = [coder decodeFloatForKey: @"NSGrid_leadingPadding"];
+            }
+          if ([coder containsValueForKey: @"NSGrid_owningGrid"])
+            {
+              _gridView = [coder decodeObjectForKey: @"NSGrid_owningGrid"]; // weak
+            }
+          if ([coder containsValueForKey: @"NSGrid_trailingPadding"])
+            {
+              _trailingPadding = [coder decodeFloatForKey: @"NSGrid_trailingPadding"];
+            }
+          if ([coder containsValueForKey: @"NSGrid_width"])
+            {
+              _width = [coder decodeFloatForKey: @"NSGrid_width"];
+              NSDebugLog(@"_width = %f", _width);
+            }
+          if ([coder containsValueForKey: @"NSGrid_xPlacement"])
+            {
+              _xPlacement = [coder decodeIntegerForKey: @"NSGrid_xPlacement"];
+            }      
+        }
+      else
+        {
+          [coder decodeValueOfObjCType: @encode(BOOL)
+                                    at: &_isHidden];
+          [coder decodeValueOfObjCType: @encode(CGFloat)
+                                    at: &_leadingPadding];
+          _gridView = [coder decodeObject]; 
+          [coder decodeValueOfObjCType: @encode(CGFloat)
+                                    at: &_trailingPadding];
+          [coder decodeValueOfObjCType: @encode(CGFloat)
+                                    at: &_width];
+          [coder decodeValueOfObjCType: @encode(NSInteger)
+                                    at: &_xPlacement];
+        }
+    }
+  return self;
+}
+
+@end
+
+/// Row ///
+@implementation NSGridRow
+
++ (void) initialize
+{
+  if (self == [NSGridRow class])
+    {
+      [self setVersion: 1];
+    }
+}
+
+- (void) setGridView: (NSGridView *)gridView
+{
+  _gridView = gridView; // weak reference...
+}
+
+- (NSGridView *) gridView
+{
+  return _gridView;
+}
+
+- (NSInteger) numberOfCells
+{
+  NSUInteger ri = [_gridView indexOfRow: self];
```

Comment:
Same question as above

Reply:
Since they can be merged, I didn't want to rely on [NSGridView numberOfColumns/Rows] for this data.   While this way of doing it is not as fast it is more accurate if the cells are merged.  Currently the code doesn't support merging, but when it does I, hopefully, won't need to revisit these methods.

Reply:
Again, take a look at cellAtColumnIndex:rowIndex:. This method would fail if your argument would be true.

Reply:
Tested on macOS and changed implementation.